src/SFA.DAS.EmployerIncentives.Commands/ApprenticeshipIncentive/EmploymentCheck/RefreshEmploymentChecksCommandHandler.cs

```diff
@@ -0,0 +1,55 @@
+﻿using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.EntityFrameworkCore.Internal;
+using SFA.DAS.EmployerIncentives.Abstractions.Commands;
+using SFA.DAS.EmployerIncentives.Commands.Persistence;
+using SFA.DAS.EmployerIncentives.Commands.Types.ApprenticeshipIncentive;
+using SFA.DAS.EmployerIncentives.Domain.Interfaces;
+
+namespace SFA.DAS.EmployerIncentives.Commands.ApprenticeshipIncentive.EmploymentCheck
+{
+    public class RefreshEmploymentChecksCommandHandler : ICommandHandler<RefreshEmploymentChecksCommand>
+    {
+        private readonly IApprenticeshipIncentiveDomainRepository _incentiveDomainRepository;
+        private readonly ICommandPublisher _commandPublisher;
+        private readonly ICollectionCalendarService _collectionCalendarService;
+
+        public RefreshEmploymentChecksCommandHandler(IApprenticeshipIncentiveDomainRepository incentiveDomainRepository, 
+                                                     ICommandPublisher commandPublisher,
+                                                     ICollectionCalendarService collectionCalendarService)
+        {
+            _incentiveDomainRepository = incentiveDomainRepository;
+            _commandPublisher = commandPublisher;
+            _collectionCalendarService = collectionCalendarService;
+        }
+        public async Task Handle(RefreshEmploymentChecksCommand command, CancellationToken cancellationToken = default(CancellationToken))
+        {
+            if (await ActivePeriodInProgress())
```

Comment:
Use the IPeriodEndIncompatible on the command so that the decorator implements this functionality so this is not duplicated and littered through out the code base but is maintained in one place.